doc/content/source/actions/SetupMGXSAction.md

```diff
@@ -0,0 +1,206 @@
+# SetupMGXSAction
+
+!alert note
+`SetupMGXSAction` can only set up multi-group cross section generation in problems which contain a
+[OpenMCCellAverageProblem](OpenMCCellAverageProblem.md) in the `[Problem]` block. Otherwise,
+attempting to add multi-group cross sections will result in an error.
+
+## Overview
+
+The `SetupMGXSAction` action is responsible for automating the generation of multi-group cross sections (MGXS)
+on Cardinal's mesh mirror for coupled deterministic codes. The type of spatial discretization used for MGXS
+homogenization is selected by setting `tally_type` to `cell` (to select a [distributed cell tally](CellTally.md))
+or to `mesh` ([to select an unstructured mesh tally](MeshTally.md)). The energy groups boundaries used can either be
+specified in `energy_boundaries`, or a common group structure can be selected by setting `group_structure`. The type of
+particle to filter for can be selected in `particle`. At present valid options include `neutron` and `photon` cross
```

Comment:
I'd remove the sentence "At present valid" - I think we can encapsulate the options in the enumeration docs automatically, and I'm not aware of any plans to add charged particle transport to OpenMC (don't want to give readers false anticipation).